drv/update-api/src/lib.rs

```diff
@@ -81,20 +87,61 @@ pub enum UpdateError {
     // Specific to RoT (LPC55)
     SpRotError = 21,
 
+    #[idol(server_death)]
+    ServerRestarted = 22,
+
     // Caboose checks
     ImageBoardMismatch = 23,
     ImageBoardUnknown = 24,
 
-    #[idol(server_death)]
-    ServerRestarted = 22,
+    NotImplemented = 25,
 
-    Unknown = 0xff,
+    Unknown = 0xff, // XXX With Hubpack encoding, inserting a new code above this will change the deserializer's notion of Unknown.
 }
 
 impl hubpack::SerializedSize for UpdateError {
     const MAX_SIZE: usize = core::mem::size_of::<UpdateError>();
 }
 
+#[repr(u8)]
+#[derive(
+    Eq,
+    FromPrimitive,
+    AsBytes,
+    PartialEq,
+    Clone,
+    Copy,
+    Serialize,
+    Deserialize,
+    SerializedSize,
+)]
+pub enum ResetIntent {
+    Normal = 1,
+    Persistent,
+    Transient,
+    ExpensiveAndIrrevocableProdToDev = 86,
+}
+
+impl From<gateway_messages::ResetIntent> for crate::ResetIntent {
+    fn from(value: gateway_messages::ResetIntent) -> Self {
+        match value {
+            gateway_messages::ResetIntent::Normal => Self::Normal,
+            gateway_messages::ResetIntent::Persistent => Self::Persistent,
+            gateway_messages::ResetIntent::Transient => Self::Transient,
+            gateway_messages::ResetIntent::ExpensiveAndIrrevocableProdToDev => {
+                Self::ExpensiveAndIrrevocableProdToDev
+            }
+        }
+    }
+}
+
+#[derive(Clone, Copy, SerializedSize, Serialize, Deserialize)]
+#[repr(C)]
+pub struct ResetComponentHeader {
+    pub intent: ResetIntent,
+    pub target: UpdateTarget,
```

Comment:
The `UpdateTarget` enumeration contains a lot of options. Currently these are:

```rust
    // Represents targets where we only ever write to a single
    // alternate flash location. This is typically used in
    // conjunction with a bank swap feature.
    Alternate = 1,
    // Represents targets where we must write to a specific range
    // of flash.
    ImageA = 2,
    ImageB = 3,
    Bootloader = 4,
    // For testing
    DevNull = 0xff,
```

Is this the type you intended? If so, what does it mean to reset a bootloader, or devnull? What is the difference between sending ImageA and Alternate? If only one or the other is permitted, how do clients know which to choose when talking to an update service?

What does it mean to issue a reset of `Normal` with a particular image named in the `target` field?